doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,356 @@
+
+And for this moment our stream is ready to be launched! An example of the
+Streaming loop:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp str_loop
+
+There are some other useful possibilities provided, e.g. you can abort stream
+processing via `cv::GStreamingCompiled::stop()` method (the next call of
+`cv::GStreamingCompiled::start()` will start processing the stream from the very
+beginning).
+
+### 3) ??? {#gapi_fb_s_3}
+
+### 4) You are breathtaking! {#gapi_fb_s_4}
+
+Now, when all the new G-API features are described and understood, we can
+discuss some algorithms.
+
+# Face Beautification algorithm {#gapi_fb_algorithm}
+
+The algorithm of a face retouching has been chosen as a subject for this
+tutorial to discover G-API usage convenience and efficiency in the context of
+solving a real CV problem. Its main idea is to implement a real-time videostream
+processing that detects faces and applies some filters to make them look
+beautiful (more or less). The pipeline is the following:
+
+![The Face Beautification algorithm](pics/algo.png)
+
+The algorithm consists of two parts: an inference of two networks (including
+data pre- and post-processing) and an image filtering pipeline which uses the
+inference data (creating masks, applying filters and composing the output
+image).
+
+The details of obvious operations (like command line parsing or single-string
+kernels implementing) are not going to be described in this text; there will be
+only comments about debatable features implementation.
+
+Let's start at the begining of a pipline graph definition.
+
+## Face detector inference and post-processing {#gapi_fb_face_detect}
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp gar_ROI
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp fd_inf
+
+The `config::smth` variables are constants defined in the `config` namespace:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp config
+
+After the inference, output `cv::Mat` post-processing is required: the
+face detector returns a blob with the shape [1, 1, N, 7], where N is
+the number of bounding boxes containing detected faces. Structure of an output
+for every face is the following:
+`[image_id, label, conf, x_min, y_min, x_max, y_max]`;
+all the seven elements are floating point. For our further goals we need to take
+only results with `image_id > 0` because a negative value in this field
+indicates the end of detections; also we can cut detections by the `conf` field
+to avoid detector's mistakes; so the Mat parsing looks like this code:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp vec_ROI
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp fd_pp
+
+Some points to be mentioned about this kernel:
+
+- There is `cv::Mat` data parsing by the pointer on a float used.
```

Comment:
Is it better: the pointer on a float -> `*float`?

Reply:
Do you need this sentence at all?

Reply:
don't know, only if this point is interesting to be mentioned. Is it?